shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.mm

```diff
@@ -1014,40 +1307,134 @@ - (void)testTextChangesDoNotTriggerUpdateEditingClientWithDelta {
+    CFRunLoopRunInMode((__bridge CFStringRef)runLoopMode, 0, true);
+  }
   XCTAssertEqual(updateCount, 1);
+  XCTAssertTrue(done);
 
   // Text changes don't trigger an update.
   XCTAssertEqual(updateCount, 1);
   [inputView setTextInputState:@{@"text" : @"AFTER"}];
+  done = false;
+  XCTAssertFalse(done);
+  CFRunLoopPerformBlock(CFRunLoopGetMain(), (__bridge CFStringRef)runLoopMode, ^{
+    done = true;
+  });
+
+  while (!done) {
+    // Each invocation will handle one source.
+    CFRunLoopRunInMode((__bridge CFStringRef)runLoopMode, 0, true);
+  }
   XCTAssertEqual(updateCount, 1);
+  XCTAssertTrue(done);
+
   [inputView setTextInputState:@{@"text" : @"AFTER"}];
+  done = false;
+  XCTAssertFalse(done);
+  CFRunLoopPerformBlock(CFRunLoopGetMain(), (__bridge CFStringRef)runLoopMode, ^{
+    done = true;
+  });
+
+  while (!done) {
+    // Each invocation will handle one source.
+    CFRunLoopRunInMode((__bridge CFStringRef)runLoopMode, 0, true);
+  }
   XCTAssertEqual(updateCount, 1);
+  XCTAssertTrue(done);
 
   // Selection changes don't trigger an update.
   [inputView
       setTextInputState:@{@"text" : @"SELECTION", @"selectionBase" : @0, @"selectionExtent" : @3}];
+  done = false;
+  XCTAssertFalse(done);
+  CFRunLoopPerformBlock(CFRunLoopGetMain(), (__bridge CFStringRef)runLoopMode, ^{
+    done = true;
+  });
+
+  while (!done) {
+    // Each invocation will handle one source.
+    CFRunLoopRunInMode((__bridge CFStringRef)runLoopMode, 0, true);
+  }
   XCTAssertEqual(updateCount, 1);
+  XCTAssertTrue(done);
+
   [inputView
       setTextInputState:@{@"text" : @"SELECTION", @"selectionBase" : @1, @"selectionExtent" : @3}];
+  done = false;
+  XCTAssertFalse(done);
+  CFRunLoopPerformBlock(CFRunLoopGetMain(), (__bridge CFStringRef)runLoopMode, ^{
+    done = true;
+  });
+
+  while (!done) {
+    // Each invocation will handle one source.
+    CFRunLoopRunInMode((__bridge CFStringRef)runLoopMode, 0, true);
+  }
   XCTAssertEqual(updateCount, 1);
+  XCTAssertTrue(done);
 
   // Composing region changes don't trigger an update.
   [inputView
       setTextInputState:@{@"text" : @"COMPOSING", @"composingBase" : @1, @"composingExtent" : @2}];
+  done = false;
+  XCTAssertFalse(done);
+  CFRunLoopPerformBlock(CFRunLoopGetMain(), (__bridge CFStringRef)runLoopMode, ^{
+    done = true;
+  });
+
+  while (!done) {
+    // Each invocation will handle one source.
+    CFRunLoopRunInMode((__bridge CFStringRef)runLoopMode, 0, true);
+  }
   XCTAssertEqual(updateCount, 1);
+  XCTAssertTrue(done);
+
   [inputView
       setTextInputState:@{@"text" : @"COMPOSING", @"composingBase" : @1, @"composingExtent" : @3}];
+  done = false;
+  XCTAssertFalse(done);
+  CFRunLoopPerformBlock(CFRunLoopGetMain(), (__bridge CFStringRef)runLoopMode, ^{
+    done = true;
+  });
+
+  while (!done) {
+    // Each invocation will handle one source.
+    CFRunLoopRunInMode((__bridge CFStringRef)runLoopMode, 0, true);
+  }
+  XCTAssertTrue(done);
```

Comment:
Would it be any cleaner to abstract this code to a function or something?  It's kind of a bummer that we have to clear the runloop all over the place like this, but it makes sense.  The next time a test is added here, it should be obvious that this is necessary since it's all over.